src/prefect/tasks/snowflake/snowflake.py

```diff
@@ -29,6 +29,8 @@ class SnowflakeQuery(Task):
             takes snowflake AUTOCOMMIT parameter
         - cursor_type (SnowflakeCursor, optional): specify the type of database
             cursor to use for the query, defaults to SnowflakeCursor
+        - authenticator (str, optional): type of authenticator to use for initiating
+            connection (oauth, externalbrowser...), default to None
```

Comment:
Can you include a link to the Snowflake documentation so that users of this task know what authenticator options are available? I think this link should work: https://docs.snowflake.com/en/user-guide/python-connector-api.html#connect. 

Also, it's important to note that `externalbrowser` and Okta based access will only work in environments where a browser is available. 

It seems for the `oauth` authenticator a token needs to be passed to the Snowflake connection. Can you add an optional token parameter to these tasks that is only required when `authenticator` is set to `oauth`?

Reply:
Good call I added this parameter as optional